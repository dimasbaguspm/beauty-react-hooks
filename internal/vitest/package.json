{
  "name": "@react-beauty/vitest",
  "version": "0.0.0",
  "private": true,
  "license": "MIT",
  "type": "module",
  "main": "./src/index.ts",
  "bin": "../../node_modules/.bin/vitest",
  "exports": {
    ".": "./src/index.ts",
    "./setting": "./src/base.ts",
    "./setup-file": "./src/setup-file.ts"
  },
  "scripts": {
    "lint": "eslint --fix",
    "check": "tsc --noEmit"
  },
  "devDependencies": {
    "@react-beauty/eslint": "workspace:*",
    "@react-beauty/tsc": "workspace:*",
    "@testing-library/dom": "10.4.0",
    "@testing-library/jest-dom": "6.6.3",
    "@testing-library/react": "16.3.0",
    "jsdom": "26.0.0",
    "vitest": "3.1.1"
  }
}
