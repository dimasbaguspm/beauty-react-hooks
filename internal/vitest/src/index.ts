import { resolve } from "node:path";
import { fileURLToPath } from "node:url";
import type { ViteUserConfig } from "vitest/config";

const rootDir = fileURLToPath(new URL(".", import.meta.url));

export const vitestConfig = (
  config: ViteUserConfig["test"],
): ViteUserConfig => ({
  test: {
    globals: true,
    environment: "jsdom",
    include: ["src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}"],
    setupFiles: [resolve(rootDir, "setup-file.ts")],
    passWithNoTests: true,
    ...config,
  },
});

export * from "vitest/config";
