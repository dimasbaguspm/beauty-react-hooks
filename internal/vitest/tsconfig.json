{
  "extends": "@react-beauty/tsc/base.json",
  "compilerOptions": {
    "types": [
      "@testing-library/jest-dom"
    ]
  },
  "include": [
    "src/**/*"
  ],
  "exclude": [
    "node_modules",
    "dist"
  ]
}
