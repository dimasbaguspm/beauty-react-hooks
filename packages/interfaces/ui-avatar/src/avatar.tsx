import { FC, ReactNode } from "react";

export interface AvatarImageProps {
  src: string;
}

export const AvatarImage: FC<AvatarImageProps> = () => {
  return <div>Avatar Image</div>;
};

export interface AvatarInitialProps {
  children: ReactNode;
}

export const AvatarInitial: FC<AvatarInitialProps> = () => {
  return <div>bar</div>;
};
