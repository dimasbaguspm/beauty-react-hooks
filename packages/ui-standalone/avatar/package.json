{
  "name": "@react-beauty/ui-avatar",
  "author": "Dimas Bagus P <dimas.bagus.pm1@gmail.com> (https://dimasbaguspm.github.io)",
  "version": "1.0.2",
  "description": "Collection user interface avatar from @react-beauty",
  "main": "./src/index.ts",
  "publishConfig": {
    "access": "public",
    "main": "./dist/index.js",
    "type": "./dist/index.d.ts"
  },
  "type": "module",
  "scripts": {
    "lint": "eslint src",
    "check": "tsc --noEmit",
    "build": "vite build"
  },
  "files": [
    "dist"
  ],
  "keywords": [
    "react",
    "react-hooks",
    "react-beauty",
    "use-clipboard"
  ],
  "license": "MIT",
  "peerDependencies": {
    "@types/react": "*",
    "react": "^16.8 || ^17.0 || ^18.0 || ^19.0 || ^19.0.0-rc"
  },
  "devDependencies": {
    "@react-beauty/eslint": "workspace:*",
    "@react-beauty/tsconfig": "workspace:*",
    "eslint": "9.24.0",
    "typescript": "5.8.3",
    "vite": "6.2.0",
    "vite-plugin-dts": "4.5.3"
  }
}
