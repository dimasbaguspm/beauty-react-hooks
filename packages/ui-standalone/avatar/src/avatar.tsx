import { FC } from "react";
import styles from "./avatar.module.css";

export const AvatarImage: FC = () => {
  return <div className={styles.rbAvatar}>Avatar Image</div>;
};

export const AvatarInitial: FC = () => {
  return <div>Avatar Initial</div>;
};
