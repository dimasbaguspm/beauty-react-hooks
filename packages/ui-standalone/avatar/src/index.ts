export { AvatarImage, AvatarInitial } from "./avatar";
