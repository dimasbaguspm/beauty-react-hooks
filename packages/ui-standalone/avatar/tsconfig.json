{
  "extends": "@react-beauty/tsconfig/react.json",
  "compilerOptions": {
    "outDir": "dist/types",
    "allowJs": true,
    "emitDeclarationOnly": true,
    "declaration": true,
    "declarationMap": false,
    "allowArbitraryExtensions": true
  },
  "include": [
    "src/**/*"
  ],
  "exclude": [
    "node_modules",
    "dist"
  ]
}
