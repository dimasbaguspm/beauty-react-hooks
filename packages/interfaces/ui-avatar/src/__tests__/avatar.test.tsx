import { render } from "@react-beauty/vitest/selector";

import { AvatarInitial, AvatarPicture } from "../avatar";

describe("Avatar", () => {
  it("should work as expected", () => {
    expect(
      render(<AvatarInitial>bar</AvatarInitial>).asFragment(),
    ).toMatchSnapshot();
  });

  it("should work as expected", () => {
    expect(
      render(<AvatarPicture size="2xl" src="foo" />).asFragment(),
    ).toMatchSnapshot();
  });
});
