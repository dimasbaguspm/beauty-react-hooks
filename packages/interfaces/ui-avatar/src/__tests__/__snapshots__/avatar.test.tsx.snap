// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > should work as expected 1`] = `
<DocumentFragment>
  <div>
    bar
  </div>
</DocumentFragment>
`;
